iceoryx2-bb/cxx/BUILD.bazel

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2025 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache Software License 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+# which is available at https://opensource.org/licenses/MIT.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+)
+
+cc_library(
+    name = "iceoryx2-bb-containers-cxx",
+    hdrs = glob(["include/**"]),
+    strip_include_prefix = "include",
+    deps = [
+        "//:iceoryx2-c-shared",
+        "@iceoryx//:iceoryx_hoofs",
```

Comment:
```suggestion
```

Those are not dependencies and hopefully will never be. Maybe iceoryx_hoofs will temporaily become a dependency but ideally we can avoid it.

Eventually, this shall become a library without dependencies, except `libc` and the `STL`. Long term potentially even without the `STL`.